src/state/application/updater.ts

```diff
@@ -119,7 +121,14 @@ export default function Updater(): null {
   }, [dispatch, debouncedState.chainId])
 
   useEffect(() => {
-    if (!account || !library?.provider?.request || !library?.provider?.isMetaMask) {
+    const isCbWalletDappBrowser = window?.ethereum?.isCoinbaseWallet
+    const isWalletlinkOrInjectedConnector =
+      connector instanceof WalletLinkConnector || connector instanceof InjectedConnector
+    const isWalletlink =
+      !!window?.WalletLinkProvider || (!!window?.walletLinkExtension && isWalletlinkOrInjectedConnector)
+    const isCbWallet = isCbWalletDappBrowser || isWalletlink
+    const isMetamaskOrCbWallet = library?.provider?.isMetaMask || isCbWallet
+    if (!account || !library?.provider?.request || !isMetamaskOrCbWallet) {
```

Comment:
it looks like this code actually has a bug... if you have a wallet that supports EIP 3085 and then switch to one that doesn't, it looks like `implements3085` will have last been set to true